File: sample.java
// Define the package
package com.example.animal;

// Enum to represent different animal types
enum AnimalType {
    DOG, CAT, BIRD
}

// Class to represent an Animal
public class Animal {
    // Variables (fields) for the Animal class
    private String name;
    private AnimalType type;
    private int age;

    // Constructor for the Animal class
    public Animal(String name, AnimalType type, int age) {
        this.name = name;
        this.type = type;
        this.age = age;
    }

    // Method to display details of the Animal
    public void displayInfo() {
        System.out.println("Name: " + name);
        System.out.println("Type: " + type);
        System.out.println("Age: " + age);
    }

    // Method to make the animal "speak" based on its type
    public void makeSound() {
        switch (type) {
            case DOG:
                System.out.println(name + " says: Woof!");
                break;
            case CAT:
                System.out.println(name + " says: Meow!");
                break;
            case BIRD:
                System.out.println(name + " says: Tweet!");
                break;
            default:
                System.out.println(name + " makes a sound.");
        }
    }

    // Main method to run the program
    public static void main(String[] args) {
        // Create a few Animal objects
        Animal dog = new Animal("Rex", AnimalType.DOG, 3);
        Animal cat = new Animal("Whiskers", AnimalType.CAT, 2);

        // Display information and sounds for each animal
        dog.displayInfo();
        dog.makeSound();
        System.out.println(); // For better readability

        cat.displayInfo();
        cat.makeSound();
    }
}

File: sample.py
# Define a class called 'Person'
class Person:
    # Constructor to initialize a person's name and age
    def __init__(self, name, age):
        self.name = name
        self.age = age

    # Method to greet the person
    def greet(self):
        print(f"Hello, my name is {self.name} and I am {self.age} years old.")

    # Method to check if the person is an adult
    def is_adult(self):
        if self.age >= 18:
            return True
        else:
            return False

# Function to display whether a person is an adult or not
def check_adult(person):
    if person.is_adult():
        print(f"{person.name} is an adult.")
    else:
        print(f"{person.name} is not an adult.")

# Create objects of the Person class
person1 = Person("Alice", 25)
person2 = Person("Bob", 16)

# Call the greet method for each person
person1.greet()
person2.greet()

# Check if the persons are adults
check_adult(person1)
check_adult(person2)
